src/roadmap/common.py

```diff
@@ -162,6 +225,22 @@ async def get_allowed_host_groups(
     return allowed_group_ids
 
 
+if settings.use_kessel:
+    get_allowed_host_groups = get_allowed_host_groups_kessel
+else:
+    get_allowed_host_groups = get_allowed_host_groups_rbac
+
+# async def get_allowed_host_groups(
+#     settings: t.Annotated[Settings, Depends(Settings.create)],
+#     rbac: t.Annotated[set[str | None], Depends(get_allowed_host_groups_rbac)],
+#     kessel: t.Annotated[set[str | None], Depends(get_allowed_host_groups_kessel)],
+# ) -> set[str | None]:
+#     if settings.use_kessel:
+#         return kessel
+#
+#     return rbac
```

Comment:
I'm unfamiliar with FastAPI, let me know if there is a better pattern for this.

I'm trying to have a conditional dependency - I tried what is commented and didn't seem to work (but happy to try again if that's the right pattern, maybe i missed something)

So what I ended up doing is to select the dependencies at loading/startup time